internal/mempool/mempool.go

```diff
@@ -205,11 +204,12 @@ func (txmp *TxMempool) TxsAvailable() <-chan struct{} {
 	return txmp.txsAvailable
 }
 
-// CheckTx executes the ABCI CheckTx method for a given transaction. It acquires
-// a read-lock attempts to execute the application's CheckTx ABCI method via
-// CheckTxAsync. We return an error if any of the following happen:
+// CheckTx executes the ABCI CheckTx method for a given
+// transaction. It acquires a read-lock attempts to execute the
+// application's CheckTx ABCI method via. We return an error if any of
+// the following happen:
```

Comment:
Nits: Missing conjunction (pre-existing), finish phrase.

```suggestion
// CheckTx executes the ABCI CheckTx method for a given transaction.
// It acquires a read-lock and attempts to execute the application's
// CheckTx ABCI method synchronously. We return an error if any of
// the following happen:
```